lms/services/course.py

```diff
@@ -38,15 +55,52 @@ def any_with_setting(self, group, key, value=True):
             .count()
         )
 
-    def _get(self, authority_provided_id):
+    def _get_legacy(self, authority_provided_id):
         return self._db.query(LegacyCourse).get(
             (self._consumer_key, authority_provided_id)
         )
 
-    def _create(self, authority_provided_id):
+    def _get(self, authority_provided_id):
+        return (
+            self._db.query(Course)
+            .filter_by(
+                application_instance=self._application_instance,
+                authority_provided_id=authority_provided_id,
+            )
+            .one_or_none()
+        )
+
+    def _create_legacy(self, authority_provided_id):
+        course = LegacyCourse(
+            consumer_key=self._consumer_key,
+            authority_provided_id=authority_provided_id,
+            settings=self._new_course_settings(authority_provided_id),
+        )
+
+        self._db.add(course)
+
+        return course
+
+    def _create(
+        self, authority_provided_id, context_id, name, extra, settings=None
+    ):  # pylint: disable=too-many-arguments
+        course = Course(
+            application_instance_id=self._application_instance.id,
+            authority_provided_id=authority_provided_id,
+            lms_id=context_id,
+            lms_name=name,
+            settings=settings or self._new_course_settings(authority_provided_id),
```

Comment:
For creating new `Courses` we might be creating this from an existing one so it take optional settings but defaults to create new ones from the AI